### PR TITLE
feat(molecule/dataCounter): improve datacounter improving features

### PR DESCRIPTION
**Pull request by AitorRodriguez990**
## MOLECULE/DATACOUNTER
**TASK**: [SCMI-50178](https://jira.scmspain.com/browse/SCMI-50178)

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Tasks

- [x] Add loading state on buttons action.
- [x] Change literals to icons on `-` and `+` to be more attractive.
- [x] Fix buttons action size to show them on `small` if component receives this prop.
- [x] Be able to disable only the `input` field.
- [x] Use JS demo instead of Playground to improve demonstration.

### Description, Motivation and Context

In order to improve `DataCounter` to be more attractive and useful, we add some features. Because no one vertical use this component, we change it with freedom.

### Screenshots - Animations

image

### CODE REVIEW

**Comment by andresin87** [comment]
can't resolve '@s-ui/react-molecule-field'

build n test processes failed

rebuilding...

File: components/molecule/dataCounter/src/index.js [hunk 1]
```diff
@@ -2,36 +2,43 @@ import {useState} from 'react'
 import PropTypes from 'prop-types'
 import cx from 'classnames'
 
-import AtomButton from '@s-ui/react-atom-button'
+import AtomButton, {atomButtonSizes} from '@s-ui/react-atom-button'
 import AtomInput, {inputSizes} from '@s-ui/react-atom-input'
 import MoleculeField from '@s-ui/react-molecule-field'
+import AddIcon from '@adv-ui/fc-iconset/lib/Add'
```
Comment (andresin87) [inline components/molecule/dataCounter/src/index.js]: Icon might be customizable by consumers of this component. Next line...same comment

**Review by andresin87 — CHANGES_REQUESTED** [review]

File: components/molecule/dataCounter/src/index.js (hunk 1 above)
Comment (AitorRodriguez990) [reply components/molecule/dataCounter/src/index.js]: Fixed 👍

**Review by danivm — APPROVED** [review]

**Review by emiliovillu — APPROVED** [review]

**Review by adriadoz — APPROVED** [review]

**Review by ericdarbra — APPROVED** [review]

File: demo/molecule/dataCounter/demo/package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "@adv-ui/fc-iconset": "3",
+    "@s-ui/react-molecule-data-counter": "1"
```
Comment (andresin87) [inline demo/molecule/dataCounter/demo/package.json]: This package might be imported from components directory (and it is), not the published. If its not you cant show demo features until it's already merged. This dependency is not necesary

**Pull request merged**